src/libraries/Microsoft.PowerFx.Connectors/Public/CdpRecordType.cs

```diff
@@ -59,20 +53,21 @@ public override bool TryGetFieldType(string fieldName, out FormulaType type)
                 return false;
             }
```

Comment:
we should probably not be calling this.  
[1] it can potentially lead to recursion + stackoverflows (since the dtype may end up calling back onto this RecordType). 

[2] It's creating an inconsistency for relationship fields. 
This returns the scalar type ('w'), and then it's inconsistent when we expect scalar type (returned on line 52, 60) vs. expanded relationship (line 70). 


Dataverse mitigates this with lazily expanding the Dtype on relationships.